File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SplitCollectionEditor.java
package uk.gov.justice.digital.hmpps.oauth2server.config;

import org.springframework.beans.propertyeditors.CustomCollectionEditor;

import java.util.Collection;

public class SplitCollectionEditor extends CustomCollectionEditor {

    private final Class<? extends Collection> collectionType;
    private final String splitRegex;

    public SplitCollectionEditor(final Class<? extends Collection> collectionType, final String splitRegex) {
        super(collectionType, true);
        this.collectionType = collectionType;
        this.splitRegex = splitRegex;
    }

    @Override
    public void setAsText(final String text) throws IllegalArgumentException {
        if (text.isEmpty()) {
            super.setValue(super.createCollection(this.collectionType, 0));
        } else {
            super.setValue(text.split(splitRegex));
        }
    }
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/SplitCollectionEditor.kt
package uk.gov.justice.digital.hmpps.oauth2server.config

import org.springframework.beans.propertyeditors.CustomCollectionEditor

class SplitCollectionEditor(
  private val collectionType: Class<out MutableCollection<*>?>,
  private val splitRegex: String,
) : CustomCollectionEditor(collectionType, true) {

  override fun setAsText(text: String) {
    if (text.isEmpty()) {
      super.setValue(super.createCollection(this.collectionType, 0))
    } else {
      super.setValue(text.split(splitRegex.toRegex()).toTypedArray())
    }
  }
}
